thorlcr/activities/nsplitter/thnsplitterslave.cpp

```diff
@@ -178,16 +193,19 @@ class NSplitterSlaveActivity : public CSlaveActivity, implements ISharedSmartBuf
         if (!inputPrepared)
         {
             inputPrepared = true;
+            activeOutputCount = initActiveOutputCount;
             PARENT::start();
-            remainingOutputs = activeOutputs;
             ForEachItemIn(o, outputs)
             {
                 CSplitterOutput *output = (CSplitterOutput *)outputs.item(o);
-                if (output && output->isStopped())
-                    --remainingOutputs;
+                if (output->isStopped())
```

Comment:
Yes, I think the change to avoid changing stoppedOutputs in prepare avoids the race.